### ISSUE TITLE
Drush fails to provision on a clean install

### ISSUE THREAD

**Issue by mbarcia** [hashbangcode/vlad#165]
dev branch, updated to 1.0.3, with my add-on support for Windows (which I don't think interferes) I'm getting this on a first provision attempt:

==> vlad:
==> vlad: TASK: [drupal | drush | install drush via composer] ***************************
==> vlad: changed: [192.168.100.100] => {"changed": true, "cmd": ["composer", "global", "require", "drush/drush:6.*"], "
delta": "0:00:18.860671", "end": "2015-03-06 17:49:16.260165", "failed": false, "failed_when_result": false, "rc": 0, "s
tart": "2015-03-06 17:48:57.399494", "stderr": "Changed current directory to /root/.composer\n./composer.json has been c
reated\nLoading composer repositories with package information\nUpdating dependencies (including require-dev)\n  - Insta
lling drush/drush (6.5.0)\n    Downloading: connection...\n    Downloading: 0%\n    Downloading: 5%\n    Downloading: 10
%\n    Downloading: 15%\n    Downloading: 20%\n    Downloading: 25%\n    Downloading: 30%\n    Downloading: 35%\n    Dow
nloading: 40%\n    Downloading: 45%\n    Downloading: 50%\n    Downloading: 55%\n    Downloading: 60%\n    Downloading:
70%\n    Downloading: 75%\n    Downloading: 80%\n    Downloading: 85%\n    Downloading: 90%\n    Downloading: 95%\n    D
ownloading: 100%\n    Downloading: 100%\n\nWriting lock file\nGenerating autoload files", "stdout": "", "stdout_lines":
[], "warnings": []}
==> vlad:
==> vlad: TASK: [drupal | drush | set drush to executable] ******************************
==> vlad: failed: [192.168.100.100] => {"failed": true, "path": "/home/vagrant/.composer/vendor/drush/drush", "state": "
absent"}
==> vlad: msg: file (/home/vagrant/.composer/vendor/drush/drush) is absent, cannot continue
==> vlad:
==> vlad: FATAL: all hosts have already failed -- aborting

**Comment by philipnorton42** [comment]
Sometimes Ansible can get a little stuck. Can you try running vagrant provision again and see if the error continues?

**Comment by mbarcia** [comment]
Unfortunately yes :-/

**Comment by philipnorton42** [comment]
Very odd, especially considering that we haven't changed that area recently. I've got some time this weekend so I'll be running some tests and looking into issues. I'll add this to my 'watch' list. :)

**Comment by philipnorton42** [comment]
Hi @mbarcia, I've completely rebuilt my box using 1.0.3 on my system and I have no problems at all. Could you run the following just in case your versions are off:

echo "VirtualBox `VBoxManage --version`"; vagrant --version; ansible --version; vagrant plugin list

I appreciate you might be on Windows, in which case we are just looking for the versions of stuff and the vagrant plugin list.

Here is my current setup:

VirtualBox 4.3.14r95030
Vagrant 1.6.3
ansible 1.8.4
  configured module search path = None
vagrant-cachier (0.9.0)
vagrant-login (1.0.1, system)
vagrant-share (1.1.1, system)
vagrant-triggers (0.4.1)

Thanks!

**Comment by danbohea** [comment]
I'm also not seeing this error on the current dev branch. My setup:

VirtualBox 4.3.22r98236
Vagrant 1.7.2
ansible 1.8.4
  configured module search path = None
vagrant-cachier (1.1.0)
vagrant-share (1.1.4, system)
vagrant-triggers (0.5.0)

Also, Markdown has mangled the command that Phil posted above (it won't print the VirtualBox version). Here's an unmangled version:

```
echo "VirtualBox `VBoxManage --version`"; vagrant --version; ansible --version; vagrant plugin list
```

**Comment by iAugur** [comment]
Looking at the Ansible messages above my bet is that composer has installed drush in
````
/root/.composer/vendor/drush/drush
````
not
````
/home/vagrant/.composer/vendor/drush/drush
````
For some reason in this case?

**Comment by philipnorton42** [comment]
Ah, that's a good spot @iAugur. That situation would arrive if you set the rush install command to run through sudo, because in that case you log into the box as the sudo user (which is root).

@mbarcia can I just check that you've not added a sudo flag to that command? It would be at the top of the file /vlad/playbooks/roles/drupal/tasks/drush.yml

**Comment by mbarcia** [comment]
I'm on a rush now, tomorrow I'll have more time. But this is how I've fixed drush install in my current working branch, because (I guess) of similar issues I had. This may have to do with env vars not set when ansible runs on the same machine (Windows host), but I've found this to fix it.

- name: install drush 6 via composer
  shell: chdir=/home/{{ user }} composer --no-progress --no-interaction require drush/drush:6.*
  register: composer_require
  failed_when: "composer_require.rc != 0"
  changed_when: "'Nothing to install or update' not in composer_require.stdout"
  tags: drush

It first chdirs to /home/vagrant, then sets some options to reduce unneeded verbosity.

**Comment by philipnorton42** [comment]
That seems fine to me and I've added your changes to the current dev branch (which I've also merged with the windows branch). Strange how the windows version of ansible caused this.

**Comment by mbarcia** [comment]
There's no "windows version" of ansible. To support Windows, ansible gets installed in the guest. But it is indeed curious how composer gets a executes in a different directory when ansible establishes a local connection instead of a remote connection.

**Comment by mbarcia** [comment]
The same fix needs to be further applied throughout drush.yml. Also, there is a drush parameter called "destination" that should be handy to control where the other utils like coder are downloaded and ansible-verified. Need to test it and will post a PR.

**Comment by philipnorton42** [comment]
Sorry Mariano, when I said 'windows version' I meant Ansible on windows, but having just read up on things I see now how it works.
I'll keep modifying the drush playbook.

**Comment by philipnorton42** [comment]
Right, that's pushed to the dev branch, if everything works out then let us know and we'll close this issue :)

**Comment by philipnorton42** [comment]
Issue now fixed in 1.0.4.
